### ISSUE TITLE
Copy knob state URL looses value type

### ISSUE THREAD

**Issue by Silic0nS0ldier** [storybookjs/storybook#14741]
**Describe the bug**
The knobs addon supports copying a URL with existing knob states. Some knobs may use a number as the value, however they are casted to a string when restoring knob state using the copied URL.

**To Reproduce**
TODO: Create ready-to-run reproduction repo

1. Run story, change 'Select a number' knob.
2. Copy URL from knobs panel.
3. Go to URL, expect to see error thrown.

```tsx
import { select } from '@storybook/addon-knobs';
import * as React from 'react';

export default {
  title: 'repro-test'
};

export const Repro = () => {
  const num = select(
    'Select a number',
    {
      ONE: 1,
      TWO: 2,
      THREE: 3,
    },
    1
  );

  switch (num) {
    case 1:
      return <div>one</div>;
    case 2:
      return <div>two</div>;
    case 3:
      return <div>three</div>;
    default:
      throw new Error();
  }
};

```

**System**
```
System:
  OS: macOS 10.15.7
  CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
Binaries:
  Node: 12.22.1 - /usr/local/opt/node@12/bin/node
  Yarn: 1.22.10 - /usr/local/bin/yarn
  npm: 6.14.12 - /usr/local/opt/node@12/bin/npm
Browsers:
  Chrome: 90.0.4430.85
  Edge: 90.0.818.49
  Safari: 14.1
npmPackages:
  @storybook/addon-a11y: ^6.1.21 => 6.1.21 
  @storybook/addon-actions: ^6.1.21 => 6.1.21 
  @storybook/addon-knobs: ^6.1.21 => 6.1.21 
  @storybook/addon-viewport: ^6.1.21 => 6.1.21 
  @storybook/addons: ^6.1.21 => 6.1.21 
  @storybook/react: ^6.1.21 => 6.1.21
```

**Additional context**
NA

**Comment by shilman** [comment]
FYI, we’ve released `addon-controls` in Storybook 6.0. Controls are portable, auto-generated knobs that are intended to replace `addon-knobs`, which are slated for deprecation.

- [Controls documentation](https://storybook.js.org/docs/react/essentials/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try out controls today!

**Comment by Silic0nS0ldier** [comment]
Appreciate the heads up. `addon-controls` and `addon-knobs` presumably play together fine right? (for context, there are 1110 and growing stories using `@storybook/addon-knobs` in the repo where this issue was identified, making incremental adoption a must)

**Comment by shilman** [comment]
@Silic0nS0ldier Yes, they can exist side by side, possibly even in the same story/component. But I'd probably convert them file by file, i.e. when you touch a component's stories, convert them all into controls in one shot.
